Library/Homebrew/formula_installer.rb

```diff
@@ -863,12 +864,17 @@ def link(keg)
   end
 
   def install_plist
-    return unless formula.plist
+    return unless plist = formula.plist
 
-    formula.plist_path.atomic_write(formula.plist)
-    formula.plist_path.chmod 0644
-    log = formula.var/"log"
-    log.mkpath if formula.plist.include? log.to_s
+    plist = { formula.plist_name => plist } if plist.is_a? String
```

Comment:
This indicates the Formula API is broken, I'm afraid. `brew services` (and perhaps other tools) assume the output is always a string. Need to figure out a way to make `formula.plist` not return a non-string here; perhaps with a `plists` method instead which can return multiple values through a Hash.

Reply:
Well if `formula.plist` _must_ be a string before this point then I don't see how we can even have multiple `plist do` blocks defined in the first place… because the `install_plist` method only runs once; forcing the need to loop over an array/hash of strings inside this method. Are you saying that we would have to specify `plists do` blocks inside formulae instead?

There are no other references to `formula.plist` outside of this `install_plist` method… Does `brew services` not just operate on the files this method writes?

Reply:
> Well if formula.plist must be a string before this point then I don't see how we can even have multiple plist do blocks defined in the first place

You can have multiple defined and introduce a new API to return them `formula.plists` and just have `formula.plist` always return the first one instead for external consumers. `plist do` can remain as-is but the consumption `formula.plist` API needs to not change.

> Does `brew services` not just operate on the files this method writes… and if not, why not?

No, it doesn't, you can see/test the code to see why. Given it does these I'm not totally confident other tools won't either and maintaining some degree of backwards compatibility is important. I could be convinced that nothing else except `brew services` uses this but then it would need to have a PR that lands pretty much simultaneously to this stuff which handles both the old code (until this goes out in a tagged stable release) and the new code (for those running Homebrew/brew's master branch).